provider/identityprovider/htpasswd.go

```diff
@@ -119,3 +122,74 @@ func uniqueUsernameValidator() validator.List {
 		}
 	})
 }
+
+func htPasswdUserListToStringMaps(ctx context.Context, users []HTPasswdUser,
+	resource *cmv1.IdentityProviderClient) (map[string]htpasswd.HtPasswdUserWithId, error) {
+	csUserMap := make(map[string]htpasswd.HtPasswdUserWithId)
+	get, err := resource.HtpasswdUsers().List().SendContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, user := range get.Items().Slice() {
+		csUserMap[user.Username()] = htpasswd.HtPasswdUserWithId{user.ID(), user.Username(), ""}
+	}
+	for _, user := range users {
+		csUserMap[user.Username.ValueString()] = htpasswd.HtPasswdUserWithId{csUserMap[user.Username.ValueString()].Id,
+			csUserMap[user.Username.ValueString()].Username, user.Password.ValueString()}
+	}
+	finalUserMap := make(map[string]htpasswd.HtPasswdUserWithId)
+	// Remove deleted users
+	for _, user := range users {
+		if _, ok := csUserMap[user.Username.ValueString()]; ok {
+			finalUserMap[user.Username.ValueString()] = csUserMap[user.Username.ValueString()]
+		}
+	}
+	return finalUserMap, nil
+}
+
+func UpdateHTPasswd(ctx context.Context, resource *cmv1.IdentityProviderClient, state *IdentityProviderState,
+	plan *IdentityProviderState, response *resource.UpdateResponse) {
+	if reflect.DeepEqual(state.HTPasswd.Users, plan.HTPasswd.Users) {
+		return
+	}
+	stateUserMap, err := htPasswdUserListToStringMaps(ctx, state.HTPasswd.Users, resource)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't get identity provider user ID",
+			fmt.Sprintf(
+				"Can't get identity provider user ID with identifier for "+
+					"cluster '%s': %v",
+				state.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+	planUserMap, err := htPasswdUserListToStringMaps(ctx, plan.HTPasswd.Users, resource)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't get identity provider user ID",
+			fmt.Sprintf(
+				"Can't get identity provider user ID with identifier for "+
+					"cluster '%s': %v",
+				state.Cluster.ValueString(), err,
+			),
+		)
+		return
+	}
+
+	patchParams := htpasswd.PatchParams{
+		ctx, stateUserMap, planUserMap, resource,
+		[]string{}, state.Cluster.ValueString(), response,
+	}
+
+	var ok bool
+	patchParams.RemovedUsers, ok = htpasswd.DeleteUserFromState(patchParams)
+	if !ok {
+		return
```

Comment:
Shouldn't it return an error in case of !ok?

Reply:
It already adds the error to the `Response.Diagnostics` - but i can move this to `htpasswd.go` instead

Reply:
Only weird thing is that `Response.Diagnostics` takes a `string` not an `error`